src/main/java/org/peergos/util/TraceLogger.java

```diff
@@ -247,6 +277,38 @@ private void writeLog(TraceType type, String debugDetails) {
         builder.append(new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSS").format(new Date(currentTimeMillis)) + "\t");
         builder.append(type.name() + "\t");
         builder.append(debugDetails);
-        System.out.println(builder.toString());
+        builder.append("\n");
+        String log = builder.toString();
+        // TODO(sonudoo): Turn off logging to stdout.
+        System.out.print(log);
+
+        synchronized (this) {
+            try {
+                // TODO(millerm): Rollover to a new log file after some limit.
+                output.write(log.getBytes(Charset.forName("UTF-8")));
```

Comment:
Is it too much overhead to just write them to individual files? Otherwise, the daemon process would need to maintain context of a byte offset - which is fine.

Reply:
It is. File open and close are expensive operation and can potentially lead to a performance hit on the server-side.

Can you consider clearing the logs as they are exported to the backend? I have added a TODO To have rollovers after file reaches certain size limit.

Reply:
That works. Feel free to merge